src/System.Linq.Expressions/tests/Call/CallTests.cs

```diff
@@ -226,6 +225,261 @@ public static void CallByRefMutableStructIndexWriteBack(bool useInterpreter)
+            Verify(() => Expression.Call(s_method2, arg, s_valid), exceptionType, "arg0");
+            Verify(() => Expression.Call(s_method3, arg, s_valid, s_valid), exceptionType, "arg0");
+            Verify(() => Expression.Call(s_method4, arg, s_valid, s_valid, s_valid), exceptionType, "arg0");
+            Verify(() => Expression.Call(s_method5, arg, s_valid, s_valid, s_valid, s_valid), exceptionType, "arg0");
+
+            Verify(() => Expression.Call(null, s_method1, arg), exceptionType, "arg0");
+            Verify(() => Expression.Call(null, s_method2, arg, s_valid), exceptionType, "arg0");
+            Verify(() => Expression.Call(null, s_method3, arg, s_valid, s_valid), exceptionType, "arg0");
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidArg_TestData))]
+        public static void Arg1_Invalid(Expression arg, Type exceptionType)
+        {
+            Verify(() => Expression.Call(s_method2, s_valid, arg), exceptionType, "arg1");
+            Verify(() => Expression.Call(s_method3, s_valid, arg, s_valid), exceptionType, "arg1");
+            Verify(() => Expression.Call(s_method4, s_valid, arg, s_valid, s_valid), exceptionType, "arg1");
+            Verify(() => Expression.Call(s_method5, s_valid, arg, s_valid, s_valid, s_valid), exceptionType, "arg1");
+            
+            Verify(() => Expression.Call(null, s_method2, s_valid, arg), exceptionType, "arg1");
+            Verify(() => Expression.Call(null, s_method3, s_valid, arg, s_valid), exceptionType, "arg1");
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidArg_TestData))]
+        public static void Arg2_Invalid(Expression arg, Type exceptionType)
+        {
+            Verify(() => Expression.Call(s_method3, s_valid, s_valid, arg), exceptionType, "arg2");
+            Verify(() => Expression.Call(s_method4, s_valid, s_valid, arg, s_valid), exceptionType, "arg2");
+            Verify(() => Expression.Call(s_method5, s_valid, s_valid, arg, s_valid, s_valid), exceptionType, "arg2");
+            
+            Verify(() => Expression.Call(null, s_method3, s_valid, s_valid, arg), exceptionType, "arg2");
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidArg_TestData))]
+        public static void Arg3_Invalid(Expression arg, Type exceptionType)
+        {
+            Verify(() => Expression.Call(s_method4, s_valid, s_valid, s_valid, arg), exceptionType, "arg3");
+            Verify(() => Expression.Call(s_method5, s_valid, s_valid, s_valid, arg, s_valid), exceptionType, "arg3");
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidArg_TestData))]
+        public static void Arg4_Invalid(Expression arg, Type exceptionType)
+        {
+            Verify(() => Expression.Call(s_method5, s_valid, s_valid, s_valid, s_valid, arg), exceptionType, "arg4");
+        }
+
+        private static void Verify(Action action, Type exceptionType, string paramName)
```

Comment:
Nit: "Verify" is fairly vague at the call sites.  How about "AssertThrowsArgumentException", or something like that?